examples/e2e/steps/promotion/promote_model.py

```diff
@@ -0,0 +1,145 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+
+from typing import Annotated
+
+import pandas as pd
+from config import MetaConfig
+from sklearn.metrics import accuracy_score
+
+from zenml import step
+from zenml.client import Client
+from zenml.integrations.mlflow.services import MLFlowDeploymentService
+from zenml.integrations.mlflow.steps.mlflow_deployer import (
+    mlflow_model_registry_deployer_step,
+)
+from zenml.logger import get_logger
+from zenml.model_registries.base_model_registry import ModelVersionStage
+
+logger = get_logger(__name__)
+
+model_registry = Client().active_stack.model_registry
+
+
+@step
+def promote_model(
```

Comment:
Also very interesting, I think we've never considered how to do that properly in the past. Maybe it would make sense to add a specific prebuilt step for this to `zenml.integrations.mlflow.steps` since I assume this is a common use case

Reply:
Also, have you tried whether this works actually? In particular, the two `mlflow_model_registry_deployer_step` calls? It looks a bit questionable to me but TBH I also don't know how else you could do this.

Reply:
All works - no issues, but here it will be complex to split steps. I might think of refactoring `mlflow_model_registry_deployer_step`  to call a function `mlflow_model_registry_deployer_function` for consistency and use the same function calls in promotion or other steps internally. What do you think?
Logic inside the deployer is too complex to reimplement, imho.